R/Features-aggregation.R

```diff
@@ -216,35 +236,23 @@ setMethod("aggregateFeatures", "Features",
                      "effects of missing values on data aggregation.")
         message(paste(strwrap(msg), collapse = "\n"))
     }
-
-    aggregated_assay <- aggregate_by_vector(assay_i, groupBy, fun, ...)
-    aggcount_assay <- aggregate_by_vector(assay_i, groupBy, colCounts)
-    aggregated_rowdata <- Features::reduceDataFrame(rowdata_i, rowdata_i[[fcol]],
-                                                   simplify = TRUE, drop = TRUE,
-                                                   count = TRUE)
-
+    
+    aggregated_assay <- aggregate_by_vector(m, groupBy, fun, ...)
+    aggcount_assay <- aggregate_by_vector(m, groupBy, colCounts)
+    aggregated_rowdata <- Features::reduceDataFrame(rd, rd[[fcol]],
+                                                    simplify = TRUE, drop = TRUE,
+                                                    count = TRUE)
+    
     se <- SummarizedExperiment(assays = SimpleList(assay = aggregated_assay,
                                                    aggcounts = aggcount_assay),
                                rowData = aggregated_rowdata[rownames(aggregated_assay), ])
     ## If the input objects weren't SummarizedExperiments, then try to
     ## convert the merged assay into that class. If the conversion
     ## fails, keep the SummarizedExperiment, otherwise use the
     ## converted object (see issue #78).
-    if (class_i != "SummarizedExperiment")
-        se <- tryCatch(as(se, class_i),
-                        error = function(e) se)
+    if (.class != "SummarizedExperiment")
+        se <- tryCatch(as(se, .class),
+                       error = function(e) se)
     
-    hits <- findMatches(rownames(aggregated_assay), groupBy)
-    elementMetadata(hits)$names_to <- rowdata_i[[fcol]][hits@to]
-    elementMetadata(hits)$names_from <- rownames(assay_i)[hits@to]
```

Comment:
If I understand, these lines are handled by the `addAssayLink()` call in `aggregateFeatures,Features`. That's much cleaner!

Reply:
Exactly !  You can find [those lines](https://github.com/rformassspectrometry/Features/blob/64187a92014870273a4ff889b5843dcfbc58e252/R/AssayLinks.R#L258-L263) back in the `.getHits` called through the `addAssayLink`.